File: src/main/java/fit/wenchao/http_file_server/model/JsonResult.java
package fit.wenchao.http_file_server.model;

import fit.wenchao.http_file_server.constants.RespCode;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class JsonResult {

    private Object data;

    private String code;
    private String msg;

    public static JsonResult of(Object data, RespCode respCode) {
        JsonResult jsonResult = new JsonResult();
        jsonResult.data = data;
        jsonResult.code = respCode.getCode();
        jsonResult.msg = respCode.getMsg();
        return jsonResult;
    }

    public static JsonResult of(Object data, String code, String msg) {
        JsonResult jsonResult = new JsonResult();
        jsonResult.data = data;
        jsonResult.code = code;
        jsonResult.msg = msg;
        return jsonResult;
    }

    public static JsonResult ok() {
        return new JsonResult(null, RespCode.SUCCESS.getCode(), RespCode.SUCCESS.getMsg());
    }

    public static JsonResult ok(Object data) {
        return new JsonResult(data, RespCode.SUCCESS.getCode(), RespCode.SUCCESS.getMsg());
    }
}

File: src/main/java/fit/wenchao/http_file_server/exception/JsonResult.kt
package fit.wenchao.http_file_server.exception


class JsonResult {
    var data: Any? = null
    var code: String? = null
    var msg: String? = null

    companion object {
        fun of(data: Any?, respCode: RespCode): JsonResult {
            val jsonResult = JsonResult()
            jsonResult.data = data
            jsonResult.code = respCode.getCode()
            jsonResult.msg = respCode.msg
            return jsonResult
        }

        fun of(data: Any?, code: String?, msg: String?): JsonResult {
            val jsonResult = JsonResult()
            jsonResult.data = data
            jsonResult.code = code
            jsonResult.msg = msg
            return jsonResult
        }

        fun ok(): JsonResult {
            return JsonResult().apply {
                this.data = null
                this.code = RespCode.SUCCESS.getCode()
                this.msg = RespCode.SUCCESS.msg
            }
        }

        fun ok(data: Any?): JsonResult {
            return JsonResult().apply {
                this.data = data
                this.code = RespCode.SUCCESS.getCode()
                this.msg = RespCode.SUCCESS.msg
            }
        }
    }
}
